docs/developer.md

```diff
@@ -0,0 +1,312 @@
+- [Changesets](https://github.com/changesets/changesets) for versioning and changelogs.
+
+```{note}
+Below you will see several `npm run x` commands.
+These are simply aliases for other commands, defined in the `package.json` file under "scripts".
+```
+
+### Developer workflow: myst CLI
+
+The `mystmd` libraries and command line tools are written in [TypeScript](https://www.typescriptlang.org/), and require [NodeJS and npm](https://nodejs.org) for local development.
+
+:::{warning}
+The `mystmd-py` package is a thin Python wrapper around the `mystmd` bundle that can be installed using `pip` or `conda`. If you have installed `mystmd` this way, it is recommended that you uninstall it before using the local development instructions below.
+:::
+
+To do local development, [clone the repository](https://github.com/jupyter-book/mystmd):
+
+```shell
+git clone git@github.com:jupyter-book/mystmd.git
+cd mystmd
+```
+
+Then, install dependencies via npm.
+You need to do this each time you pull from upstream, since dependency versions may change:
+
+```shell
+npm install
+```
+
+Now, build `mystmd`:
+
+```shell
+npm run build
+```
+
+Optionally, you can link the built executable as your globally installed `mystmd`:
+
+```shell
+npm run link
+```
+
+```{warning}
+The build process uses unix commands that might not work properly on Windows.
+When building on Windows, use either WSL or a unix-like shell (such as Git Bash or MSYS2), and make sure that npm is set to use these by default (`npm config set script-shell path/to/shell.exe`).
+```
+
+These commands allow you to use the `myst` CLI from any directory; source code changes are picked up after each `npm run build` (executed in the top-level source directory).
+
+### Developer workflow: myst-theme
+
+the [`myst-theme` README](https://github.com/jupyter-book/myst-theme/) provides a more detailed overview of the components of that package.
+
+Recall from the [architecture overview](#architecture) that `myst-md` is a React web application. It provides theming, and requires a separate content server for data. When developing, the steps are therefore to:
+
+1. Launch a content server
+2. Launch the `myst-theme` web application server (this is what you browse to)
+3. Run a process to monitor changes and rebuild `myst-theme`
+
+#### Content server
+
+We need some example data to test our theme against, such as [the example landing page](https://github.com/myst-examples/landing-pages) or ... Clone our data repository and start the content server:
+
+```shell
+git clone https://github.com/myst-examples/landing-pages
+cd landing-pages
+myst start --headless
+```
+
+(The `--headless` flag tells `myst` not to start up the theme server; we want to do that ourselves in the next step.)
+
+```{error}
+Can you help me with other good content site examples?
+```
+
+#### myst-theme server
+
+We now fire up the `myst-theme` React app, which will "hydrate" itself from the content server.
```

Comment:
This isn't quite how I would describe it. Hydrate in react means going from server HTML to client side react+interactivity. The app-server fetches from the json from the content-server, and serves HTML, which is hydrated on the client.